orttraining/orttraining/training_api/include/onnxruntime_training_c_api.h

```diff
@@ -190,21 +197,22 @@ struct OrtTrainingApi {
   ORT_API2_STATUS(OptimizerStep, _Inout_ OrtTrainingSession* sess,
                   _In_opt_ const OrtRunOptions* run_options);
 
-  /** \brief Registers the use of the given learning rate scheduler for the training session.
+  /** \brief Registers the use of the Linear learning rate scheduler for the training session.
   *
-  * Register a learning rate scheduler identified by the given enum with the given
+  * Register a Linear learning rate scheduler with the given
   * learning rate scheduler parameters. Optionally specify the initial learning rate
   * that should be used with this learning rate scheduler and training session.
   *
   * \param[in] sess The training session that should use the linear learning rate scheduler.
-  * \param[in] lr_scheduler_parameters Learning rate scheduler parameters struct.
+  * \param[in] warmup_step_count Warmup steps for LR warmup.
+  * \param[in] total_step_count Total step count.
   * \param[in] initial_lr The initial learning rate to be used by the training session.
   *
   * \snippet{doc} snippets.dox OrtStatus Return Value
   *
   */
-  ORT_API2_STATUS(RegisterLRScheduler, _Inout_ OrtTrainingSession* sess, _In_ void* lr_scheduler_parameters,
-                  _In_ enum OrtLRSchedulerType lr_scheduler_type, _In_opt_ const float* initial_lr);
+  ORT_API2_STATUS(RegisterLinearLRScheduler, _Inout_ OrtTrainingSession* sess, _In_ const int64_t warmup_step_count,
+                    _In_ const int64_t total_step_count, _In_ const float initial_lr);
```

Comment:
does the initial lr need to be a mandatory input? Also `const` can be avoided (since we are taking in the argument by value).

Reply:
Does it need to be optional? If they are defining a LR scheduler, it is not a lot of effort to pass in the initial LR. Regardless, the change is because it is non-trivial to pass optional float through C# APIs